### PR TITLE
Added both CPU and GPU profiling Support!!

### PR DESCRIPTION
**Pull request by harishsg99**
This PR allows users to monitor their CPU and GPU memory usage while training models using DeepSpeed

### CODE REVIEW

File: deepspeed/profiling/gpu_profiler/gpu_profiler.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import torch
+from torch.profiler import profile, record_function, ProfilerActivity
+class GPUProfiler(object):
+    def __init__(self, model):
+        self.model = Model()
+
+    def start_gpu_profiler(inputs):
+        with profile(activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA], record_shapes=True,profile_memory=True) as prof:
```
Comment (tjruwase) [inline deepspeed/profiling/gpu_profiler/gpu_profiler.py:8]: Why does this include `ProfilerActivity.CPU`?

Comment (harishsg99) [reply deepspeed/profiling/gpu_profiler/gpu_profiler.py]: Sorry I added by mistake! , I remove ProfilerActivity.CPU

**Review by harishsg99 — COMMENTED** [review]
I removed ProfilerActivity.CPU from gpu_profiler.py

**Comment by harishsg99** [comment]
Added CPU Support for lamb algorithm !

File: deepspeed/ops/lamb/lamb_cpu.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
+'''
```
Comment (tjruwase) [inline deepspeed/ops/lamb/lamb_cpu.py:1]: Is this a CPU implementation of the LAMB optimizer? It should be a separate PR, right?

Comment (harishsg99) [reply deepspeed/ops/lamb/lamb_cpu.py:1]: Yes I am sorry, I made a small mistake while creating PR so it got merged with the existing PR. I will create a new PR for this one.

Comment (tjruwase) [reply deepspeed/ops/lamb/lamb_cpu.py:1]: Got it. No worries. The cpu LAMB optimizer looks very interesting. I look forward to seeing the PR. Also could share the motivation and your experience with it in the PR? Thanks!

**Comment by tjruwase** [comment]
@harishsg99, can you add some unit tests? Thanks!

**Comment by tjruwase** [comment]
@harishsg99, is this ready for review. I notice the PR is failing formatting checks as below. 
image


Can you please see [here](https://github.com/microsoft/DeepSpeed/blob/master/CONTRIBUTING.md#prerequisites) for instructions on fixing? Thanks!

**Comment by harishsg99** [comment]
> @harishsg99, can you add some unit tests? Thanks!

Sure I will add them tomorrow thanks!

**Comment by cli99** [comment]
Hi @harishsg99, we would like to move the usage of the PyTorch profiler in DeepSpeed to the tutorial documentation. Would you like to start a PR there? It would be under https://github.com/microsoft/DeepSpeed/tree/master/docs/_tutorials and showcases how to use the PyTorch profiler API to profile different stages of DeepSpeed model training (not just the self.model(inputs). Thanks.
